### PR TITLE
IALERT-3031 Upgrade to java 17

### PR DESCRIPTION
**Pull request by mau-synopsys**
From Paulo's comment on ticket:

The following library projects have had their dependencies updated. For Java 17 we want an updated version of Gson. This requires all the following libraries to be released.
Library Project:

    integration-common [branch](https://github.com/blackducksoftware/integration-common/tree/ps-update-dependencies)
    integration-rest [branch](https://github.com/blackducksoftware/integration-rest/tree/ps-update-dependencies)
    phone-home-client [branch](https://github.com/blackducksoftware/phone-home-client/tree/ps-update-dependencies)
    integration-bdio [branch](https://github.com/blackducksoftware/integration-bdio/tree/ps-update-dependencies)
    blackduck-common-api [branch](https://github.com/blackducksoftware/blackduck-common-api/tree/ps-update-dependencies)
    blackduck-common [branch](https://github.com/blackducksoftware/blackduck-common/tree/ps-update-dependencies)

Including bug fixes:
- JsonIgnore annotation caused models to not be returned in AlertPagedModel
- Transactional annotation on UserActions created a proxy for use which did contain autowired beans since getAll was final
- Coalesce did not function as expected due to hibernate upgrade
- Check for token as part of login as a null token return still has ok status in response

### CODE REVIEW

File: src/main/java/com/synopsys/integration/alert/configuration/ApplicationConfiguration.java [hunk 1]
```diff
@@ -63,29 +60,29 @@ public PlatformTransactionManager transactionManager() {
...
+    //    @Bean
+    //    public SimpleJobLauncher jobLauncher() {
+    //        SimpleJobLauncher launcher = new SimpleJobLauncher();
+    //        try {
+    //            launcher.setJobRepository(jobRepository());
+    //        } catch (Exception ex) {
+    //            logger.error("Creating job launcher bean", ex);
+    //        }
+    //        return launcher;
+    //    }
```
Comment (ChomickiM) [inline src/main/java/com/synopsys/integration/alert/configuration/ApplicationConfiguration.java:85]: Should this commented section be removed?

File: web/src/main/java/com/synopsys/integration/alert/web/documentation/SwaggerConfiguration.java [hunk 2]
```diff
@@ -7,26 +7,15 @@
...
-import springfox.documentation.spi.DocumentationType;
-import springfox.documentation.spring.web.plugins.Docket;
-import springfox.documentation.swagger.web.UiConfiguration;
-import springfox.documentation.swagger.web.UiConfigurationBuilder;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.info.Info;
 
 @Configuration
-@EnableOpenApi
+//@EnableOpenApi
```
Comment (ChomickiM) [inline web/src/main/java/com/synopsys/integration/alert/web/documentation/SwaggerConfiguration.java:18]: is this annotation no longer needed?

File: alert-database/src/main/java/com/synopsys/integration/alert/database/audit/AuditEntryEntity.java [hunk 3]
```diff
@@ -53,7 +51,7 @@ public class AuditEntryEntity extends BaseEntity implements DatabaseEntity {
     private String errorMessage;
 
     @Lob
-    @Type(type = "org.hibernate.type.TextType")
+    //@Type(type = "org.hibernate.type.TextType")
```
Comment (psantos1113) [inline alert-database/src/main/java/com/synopsys/integration/alert/database/audit/AuditEntryEntity.java:54]: At this point can we just remove this line?

**Review by psantos1113 — APPROVED** [review]
Other than the comment to remove a commented annotation I don't have any issue with this PR.

**Review by ChomickiM — APPROVED** [review]

**Pull request merged**